src/Elastic.Apm/Agent.cs

```diff
@@ -4,7 +4,7 @@
 using Elastic.Apm.Config;
 using Elastic.Apm.DiagnosticSource;
 using Elastic.Apm.Logging;
-using Elastic.Apm.Model;
+using Elastic.Apm.Metrics;
```

Comment:
Removed the using.

No, there are no examples. I can push some.

Reply:
I think it's worth the effort if we want to give users ability to add metrics. The additional reason I asked for an example is because I was not sure how user should do add a new metric provider considering that `IMetricsProvider` is `internal` and `IMetricsCollector` doesn't have any way to add `IMetricsProvider`-s to it. Am I missing something?

Reply:
If we decide to exclude public API for metrics from this PR it might be worth amending PR's description

> Also adds public API interface to send custom metrics

Reply:
👍  done